src/OpenTelemetry.Instrumentation.Wcf/Implementation/InstrumentedDuplexChannel.cs

```diff
@@ -178,20 +178,27 @@ private IAsyncResult SendInternal(Message message, TimeSpan timeout, Func<AsyncC
     private void SendInternal(Message message, TimeSpan timeout, Action<RequestTelemetryState> executeSend)
     {
         RequestTelemetryState? telemetryState = null;
-        ContextCallback executeInChildContext = _ =>
+
+        void ExecuteInChildContext(object? unused)
         {
             telemetryState = ClientChannelInstrumentation.BeforeSendRequest(message, this.RemoteAddress?.Uri);
             RequestTelemetryStateTracker.PushTelemetryState(message, telemetryState, timeout, OnTelemetryStateTimedOut);
             executeSend(telemetryState);
-        };
+        }
+
+        var executionContext = ExecutionContext.Capture();
+        if (executionContext == null)
+        {
+            throw new InvalidOperationException("Cannot fetch execution context");
+        }
 
         try
         {
-            ExecutionContext.Run(ExecutionContext.Capture(), executeInChildContext, null);
```

Comment:
This is the part I'm not sure about. I'm going to try and find an expert on the .NET runtime team to ask about what to do here. I'll post back.

Reply:
Looking for any advice you can give : )

Reply:
Haven't nailed it down yet but I approved the PR. We're adding a throw here for `null` EC but `ExecutionContext.Run` should be throwing today in that case anyway and no user has complained about it AFAIK.

We can circle back to it if any new info comes to light.